pkg/apis/pingcap/v1alpha1/types.go

```diff
@@ -1437,3 +1441,272 @@ type IngressSpec struct {
 	// +optional
 	TLS []extensionsv1beta1.IngressTLS `json:"tls,omitempty"`
 }
+
+// +k8s:openapi-gen=true
+// DMCluster is the control script's spec
+type DMCluster struct {
+	metav1.TypeMeta `json:",inline"`
+	// +k8s:openapi-gen=false
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec defines the behavior of a dm cluster
+	Spec DMClusterSpec `json:"spec"`
+
+	// +k8s:openapi-gen=false
+	// Most recently observed status of the dm cluster
+	Status DMClusterStatus `json:"status"`
+}
+
+// +k8s:openapi-gen=true
+// DMDiscoverySpec contains details of Discovery members for dm
+type DMDiscoverySpec struct {
+	corev1.ResourceRequirements `json:",inline"`
```

Comment:
Suggest not syncing Discovery component, just connect to the discovery of the tidb cluster.